datamap/mv/plan/src/main/scala/org/apache/carbondata/mv/plans/modular/Harmonizer.scala

```diff
@@ -36,17 +36,33 @@ abstract class Harmonizer(conf: SQLConf)
   def batches: Seq[Batch] = {
     Batch(
       "Data Harmonizations", fixedPoint,
-      HarmonizeDimensionTable,
-      HarmonizeFactTable) :: Nil
+      Seq( HarmonizeDimensionTable) ++
+      extendedOperatorHarmonizationRules: _*) :: Nil
+//      HarmonizeFactTable) :: Nil
   }
+
+  /**
+   * Override to provide additional rules for the modular operator harmonization batch.
+   */
+  def extendedOperatorHarmonizationRules: Seq[Rule[ModularPlan]] = Nil
+}
+
+/**
+ * A full Harmonizer - harmonize both fact and dimension tables
+ */
+object FullHarmonizer extends FullHarmonizer
+
+class FullHarmonizer extends Harmonizer(new SQLConf()) {
+  override def extendedOperatorHarmonizationRules: Seq[Rule[ModularPlan]] =
+    super.extendedOperatorHarmonizationRules ++ (HarmonizeFactTable :: Nil)
 }
 
 /**
- * An default Harmonizer
+ * A semi Harmonizer - harmonize dimension tables only
```

Comment:
Can you add description for semi harmonizer?

Reply:
ok